src/scripting/StelMainScriptAPI.cpp

```diff
@@ -342,8 +342,14 @@ QStringList StelMainScriptAPI::getAllTimezoneNames()
 
 void StelMainScriptAPI::screenshot(const QString& prefix, bool invert, const QString& dir, const bool overwrite, const QString &format)
 {
+	const bool allowUnsafe = qApp->property("allow_unsafe").toBool();
+	if (!allowUnsafe && dir.isEmpty()) {
+		qWarning() << "SCRIPTING CONFIGURATION ISSUE: You are trying to save a screenshot in some dangerous path";
```

Comment:
Yes, the text was not really intended to be the final version. Do you intend to say the wording should be changed (maybe "suspect" ?) or that this check is not super useful? The main danger this way is for an attacker to achieve to overwrite some of your important images (which would not be easily achievable because the attacker would have to guess the filename or existing files) or fill the disk (which is absolutely not mitigated).

Reply:
Overwriting existing files is indeed intended when setting overwrite=true. The script this mechanism for external directories and overwriting existing files was created for is the skybox script which can be called repeatedly to overwrite skybox images which are then auto-loaded in a running (serious) game (simulation). If you really feel this is a catastrophic risk for the user's data, we can add yet another configuration option even visible in the GUI to "allow overwriting existing images when taking screenshots". This will need a longish tooltip explaining that this applies only to scripts that actually request overwriting. Manual screenshots never overwrite. This GUI checkbox should therefore best go into the scripting GUI. 

"Fill the disk" is the only relevant danger I can now identify that may affect usability of a system.  This applies to every program that can create files automatically. How does e.g. a movie renderer like Blender or movie cutting programs that can dissect a movie into single frames handle decreasing disk space? Do they output images until disk is full and get stuck with the next frame they can no longer produce? Or do they have a setting "emergency stop when disk has only ...MB free"? We could add such setting as well, and check free space before writing a screenshot.

Reply:
> Or do they have a setting "emergency stop when disk has only ...MB free"?

Sounds like overengineering for casual programs. So far I've only seen such behavior in installers. All others just write and let the errors happen (and then most report them gracefully).

Reply:
> I would not call just any external and dedicated target directory "dangerous".

Actually, the idea of this change was to prevent the script from writing outside of dedicated directories.

> Overwriting existing files is indeed intended when setting overwrite=true. The script this mechanism for external directories and overwriting existing files was created for is the skybox script which can be called repeatedly to overwrite skybox images which are then auto-loaded in a running (serious) game (simulation). If you really feel this is a catastrophic risk for the user's data, we can add yet another configuration option even visible in the GUI to "allow overwriting existing images when taking screenshots".

I am afraid my previous comment was not clear at all :) No, I do **not** believe overwriting a file is problematic. What I am not super confortable with is to be able to overwrite images to arbitrary locations (unrelated directories) because it might be possible to overwrite existing user files but this is quite far-fetched.

Reply:
far-fetched indeed. Is this not a fully constructed case? you must guess what filename exists (or maybe check for existing PNG in that directory, select a random file and overwrite it? Can we just assume users check scripts when they have downloaded them from external creators? 
I can add another tickbox in the script console. However, everybody will activate it and forget about the issue.

Reply:
...done in 5d15479fdb73ad975ce23b71539d9b90531d3736